src/EditorFeatures/TestUtilities2/Intellisense/TestState.vb

```diff
@@ -256,9 +256,9 @@ Namespace Microsoft.CodeAnalysis.Editor.UnitTests.IntelliSense
 #Enable Warning BC42358 ' Because this call is not awaited, execution of the current method continues before the call is completed
 
             Dim itemsUpdatedHandler = Sub(sender As Object, e As Data.ComputedCompletionItemsEventArgs)
-                                          ' If there is more than one item left, then it means this was the filter operation that resulted and we're done. Otherwise we know a
-                                          ' Dismiss operation is coming so we should wait for it.
-                                          If e.Items.Items.Select(Function(i) i.FilterText).Skip(1).Any() Then
+                                          ' If there is 0 or more than one item left, then it means this was the filter operation that resulted and we're done. 
+                                          ' Otherwise we know a Dismiss operation is coming so we should wait for it.
+                                          If e.Items.Items.Count() <> 1 Then
```

Comment:
this is scary.  why change this?

Reply:
n/m  i see.

Reply:
we have a special case which is when we return an empty list instead of directly dismiss the session. W/o this change the test would hang.
http://sourceroslyn.io/#Microsoft.CodeAnalysis.EditorFeatures/Implementation/IntelliSense/AsyncCompletion/ItemManager.cs,475